### ISSUE TITLE
Support S3 endpoint

### ISSUE THREAD

**Issue by andyndang** [whylabs/whylogs#238]
### Description:
```
Config file in sample notebook has property of output_path for protobuf profiles, but I can’t find that config for json and specify minio endpoint url.
```
S3 private endpoint is another use case: https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html

Reference code:
https://github.com/whylabs/whylogs/blob/c31494addb9576a48b6847c7899c6830417adb12/src/whylogs/app/writers.py#L365

`smart_open` + minio: https://github.com/RaRe-Technologies/smart_open/issues/416

```
transport_params = {
    'resource_kwargs': {
        'endpoint_url': 'http://localhost:9000',
        'aws_access_key_id':'minioadmin',
        'aws_secret_access_key':'minioadmin',
    }
}
with open('s3://data-test-bucket/test.txt', 'wb', transport_params=transport_params) as fout:
    fout.write(b'Yay I work on S3 on minio!')
```

We'll have to support `transport_params` somehow with smart_open

**Comment by kuromt** [comment]
This is one of the great feature!
Supporting s3 object store like minio is most helpfull.

**Comment by andyndang** [comment]
From Slack:

> I want to give following parameters which are not supported by current whylogs.
endpoint_url
region_name
Certification file path of s3(minio) server
Certification file path of client
Intermediate certificate file path

**Comment by lalmei** [comment]
Fixed in #277
